### PR TITLE
Add categories bar

### PR DESCRIPTION
**Pull request by fatlotus**
This adds a top bar with all categories, in order, plus a new "free" category.

Test URL: https://add-categories-bar-dot-hosted-caravel.appspot.com

### CODE REVIEW

File: caravel/templates/header.html [hunk 1]
```diff
@@ -17,7 +36,7 @@ <h1 class="branding col-xs-9 col-md-2">
     </div>
   </form>
   <div class="user-button-container col-xs-12 col-md-4">
-    <a href="{{ url_for('new_listing') }}"
+    <a href="{{ url_for('new_listing', q=request.args.get('q')) }}"
```
Comment (georgeteo) [inline caravel/templates/header.html]: Add text-decoration: none; color: black

**Comment by georgeteo** [comment]
The responsive design for the nav bar jive with the header if the page is small
screen shot 2015-10-28 at 8 52 37 am
:

File: caravel/templates/header.html [hunk 2]
```diff
@@ -1,3 +1,22 @@
+<ul class="nav nav-tabs nav-justified" id="categories">
+  {% for tag, label in categories %}
+    {% if tag in request.args.get('q', '') %}
+    <li class="active">
+      <a href="{{ url_for('search_listings',
```
Comment (georgeteo) [inline caravel/templates/header.html]: Add text-decoration: none; color: black?

File: caravel/templates/header.html [hunk 3]
```diff
@@ -1,3 +1,22 @@
...
+                          q=modify_search(remove=[tag])) }}">
+        {{ label }}
+      </a>
+    </li>
+    {% else %}
+    <li>
+      <a href="{{ url_for('search_listings',
```
Comment (georgeteo) [inline caravel/templates/header.html]: Same here.

**Comment by fatlotus** [comment]
I've changed the styling to black and added some responsiveness. Since the navigation bar shows up as a full menu, I added a JavaScript hack to scroll past it, but there's probably a cleaner way to do it.

**Comment by fatlotus** [comment]
Looks good?

**Comment by georgeteo** [comment]
Yups. Merge way.

**Pull request merged**